index.js

```diff
@@ -2,23 +2,29 @@
 
 const Promise = require('bluebird');
 const readInput = Promise.promisify(require('read'));
-const argv = require('yargs').argv;
+const argv = require('yargs').array('domain').argv;
 const xtend = require('xtend');
 const getCertificate = require('./lib.js');
 
-const read = (options) => () => {
+const read = (options) => (forceReadInput = false) => {
```

Comment:
Thanks, that makes sense. I've actually been reviewing my old code and it seems this check was unnecessary:

```js
const argument = typeof argv[options.argument] === 'string' ? argv[options.argument] : undefined
```

yargs will automatically handle type conversion (returns strings unless otherwise specified), so this should do the trick:

```js
const argument = argv[options.argument];
```

Essentially, yargs guarantees that if an option has been passed, its validator function will always be executed on a value of the correct type.